WordPress/Classes/Networking/PlansRemote.swift

```diff
@@ -54,9 +56,173 @@ private func mapPlansResponse(response: AnyObject) throws -> (activePlan: Plan,
+            inMemoryPlanFeatures = planFeatures
+            cacheDate = date
+            
+            success(planFeatures)
+            return
+        }
+        
+        fetchPlanFeatures({ [weak self] planFeatures in
+            self?.inMemoryPlanFeatures = planFeatures
+            self?.cacheDate = NSDate()
+            
+            success(planFeatures)
+        }, failure: failure)
+    }
+    
+    private var _planFeatures: PlanFeatures?
+    private var inMemoryPlanFeatures: PlanFeatures? {
+        get {
+            // If we have something in memory and it's less than a day old, return it.
+            if let planFeatures = _planFeatures,
+                let cacheDate = cacheDate where
+                cacheDateIsValid(cacheDate) {
+                    return planFeatures
+            }
+            
+            return nil
+        }
+        
+        set {
+            _planFeatures = newValue
+        }
+    }
+    
+    private func cachedPlanFeatures() -> PlanFeatures? {
+        guard let (planFeatures, _) = cachedPlanFeaturesWithDate() else { return nil}
+
+        return planFeatures
+    }
+    
+    /// @return An optional tuple containing a collection of cached plan features and the date when they were fetched
+    private func cachedPlanFeaturesWithDate() -> (PlanFeatures, NSDate)? {
+        if let cacheFileURL = cacheFileURL,
+            let path = cacheFileURL.path {
+                
+            if NSFileManager.defaultManager().fileExistsAtPath(path) {
+                do {
+                    let attributes = try NSFileManager.defaultManager().attributesOfItemAtPath(path)
+                    if let modificationDate = attributes[NSFileModificationDate] as? NSDate {
+                        if cacheDateIsValid(modificationDate) {
+                            if let response = NSData(contentsOfURL: cacheFileURL) {
+                                let json = try NSJSONSerialization.JSONObjectWithData(response, options: [])
+                                return (try mapPlanFeaturesResponse(json), modificationDate)
+                            }
+                        }
+                    }
+                }
+                catch {}
+            }
+        }
+        
+        return nil
+    }
+    
+    private func cacheDateIsValid(date: NSDate) -> Bool {
+        return NSCalendar.currentCalendar().isDateInToday(date)
+    }
+    
+    private func fetchPlanFeatures(success: PlanFeatures -> Void, failure: ErrorType -> Void) {
+        let endpoint = "plans/features"
+        let path = pathForEndpoint(endpoint, withVersion: ServiceRemoteRESTApiVersion_1_2)
+        
+        api.GET(path,
+            parameters: nil,
+            success: {
+                [weak self] requestOperation, response in
+                self?.cacheResponseData(requestOperation.responseData)
+                do {
+                    try success(mapPlanFeaturesResponse(response))
```

Comment:
I think it's probably not a big deal but if the mapping fails we'll call failure and treat it as an error, yet the data will have been cached as if it were a success.